openshift/insights-operator
# Implement Prometheus Collector pattern

This PR changes the Prometheus metrics registry/server used by IO from the `legacyregistry` to an HTTP registry served directly from the IO code. An update of the Prometheus library was included in the changes to make sure the latest version of this dependency was used.

## Manual Testing Steps

0. Start a testing cluster using the OCP image based on this PR (e.g., via cluster-bot).
0. Check in the web console that the `insights-operator` target is displayed as "up" and being properly scraped and that HTTPS is being used.
0. In the Metrics tab, try querying some of the `insightsclient_*` metrics to make sure they are being reported.

## Categories



## Sample Archive


N/A

## Documentation


N/A

## Unit Tests


N/A

## Privacy


Yes. User data are not involved in the changes.

## Changelog


## Breaking Changes


No

## References


Jira Task: https://issues.redhat.com/browse/CCXDEV-7475
